### PR TITLE
MAM-3613-show-loader-on-first-following-feed-load

### PR DESCRIPTION
**Pull request by bnolens**
[MAM-3613 : Show loader on first Following feed load](https://linear.app/theblvd/issue/MAM-3613/show-loader-on-first-following-feed-load)

### CODE REVIEW

File: Mammoth/Screens/HomeScreen/NewsFeedViewController.swift [hunk 1]
```diff
@@ -183,26 +183,30 @@ class NewsFeedViewController: UIViewController, UIScrollViewDelegate, UITableVie
...
-                guard let self else { return }
-                guard self.viewModel.snapshot.sectionIdentifiers.contains(.main) else { return }
-                if self.viewModel.snapshot.itemIdentifiers(inSection: .main).isEmpty {
-                    let type = self.viewModel.type
-                    self.viewModel.displayLoader(forType: type)
-                    
-                    Task { [weak self] in
-                        guard let self else { return }
-                        try await self.viewModel.loadLatest(feedType: type, threshold: 1)
+        DispatchQueue.main.async {
```
Comment (rileyhCode) [inline Mammoth/Screens/HomeScreen/NewsFeedViewController.swift:186]: Add a comment about why dispatching here, vs just doing it inline.

File: Mammoth/Screens/HomeScreen/NewsFeedViewController.swift [hunk 2]
```diff
@@ -898,20 +902,27 @@ extension NewsFeedViewController: NewsFeedViewModelDelegate {
...
-                loaderView.addArrangedSubview(loader)
-                self.tableView.tableFooterView = loaderView
+                if !self.isLoaderVisible() {
+                    let loaderView = UIStackView(frame: CGRect(x: 0, y: 0, width: self.tableView.frame.width, height: 40))
+                    loaderView.alignment = .center
+                    loaderView.layoutMargins = .init(top: 10, left: 10, bottom: 10, right: 10)
+                    let loader = UIActivityIndicatorView()
+                    loader.startAnimating()
+                    loaderView.addArrangedSubview(loader)
+                    loaderView.tag = 11
```
Comment (rileyhCode) [inline Mammoth/Screens/HomeScreen/NewsFeedViewController.swift:912]: Consider making the 11 a const. Not sure the best way for a one off like this. Maybe… (?)
`static let LoaderTag = 11`

**Review by rileyhCode — CHANGES_REQUESTED** [review]
A couple of small items.

File: Mammoth/Screens/HomeScreen/NewsFeedViewController.swift (hunk 1 above)
Comment (bnolens) [reply Mammoth/Screens/HomeScreen/NewsFeedViewController.swift]: This dispatch was not needed. I removed it

**Review by rileyhCode — APPROVED** [review]
LGTM

**Pull request merged**